crates/iota-replay/src/data_fetcher.rs

```diff
@@ -537,15 +552,41 @@ impl DataFetcher for RemoteFetcher {
 
         if let TransactionKind::EndOfEpochTransaction(kinds) = tx_kind_orig {
             for kind in kinds {
-                if let EndOfEpochTransactionKind::ChangeEpoch(change) = kind {
-                    // Backfill cache
-                    self.epoch_info_cache.write().put(
-                        epoch_id,
-                        (change.epoch_start_timestamp_ms, reference_gas_price),
-                    );
-
-                    return Ok((change.epoch_start_timestamp_ms, reference_gas_price));
-                }
+                let (epoch_start_timestamp_ms, reference_gas_price) = match kind {
+                    EndOfEpochTransactionKind::ChangeEpoch(change) => {
+                        let rgp = if let serde_json::Value::Object(w) = event.parsed_json {
+                            u64::from_str(&w["reference_gas_price"].to_string().replace('\"', ""))
+                                .unwrap()
```

Comment:
```suggestion
                      let rgp = if let serde_json::Value::Object(ref w) = event.parsed_json {
                            w.get("reference_gas_price")
                                .and_then(|v| {
                                    // Handle both JSON number and string representations
                                    v.as_u64()
                                        .or_else(|| v.as_str().and_then(|s| s.parse().ok()))
                                })
                                .ok_or_else(|| ReplayEngineError::UnexpectedEventFormat {
                                    event: Box::new(event.clone()),
                                })?
```
Should we use something like this to avoid unwrap()?